Signal/src/ViewControllers/ConversationView/ConversationViewItem.m

```diff
@@ -364,10 +364,15 @@ - (void)ensureViewState
                 self.messageCellType = OWSMessageCellType_DownloadingAttachment;
                 self.attachmentPointer = (TSAttachmentPointer *)attachment;
                 return;
+            } else {
+                OWSFail(@"%@ Unknown attachment type", self.tag);
             }
+        } else {
```

Comment:
TIOLI consider inverting these into a guard pattern.

Reply:
I think it's better to "fall through" in the error cases because:

* There's more than one of them.
* Each one requires the same 4-5 lines before it can exit.